hypothesis-python/src/hypothesis/internal/conjecture/datatree.py

```diff
@@ -51,12 +47,16 @@ class Branch:
     """Represents a transition where multiple choices can be made as to what
     to drawn."""
 
-    bit_length = attr.ib()
+    kwargs = attr.ib()
+    ir_type = attr.ib()
     children = attr.ib(repr=False)
 
+    # I'd really like to use cached_property here, but it requires attrs >= 23.2.0,
+    # which is almost certainly too recent for our tastes.
+    # https://github.com/python-attrs/attrs/releases/tag/23.2.0
     @property
     def max_children(self):
-        return 1 << self.bit_length
+        return compute_max_children(self.kwargs, self.ir_type)
```

Comment:
I'm worried that `compute_max_children` will be expensive, and I'd like to cache it. Since we're using a slotted attrs class, we could either (1) upgrade to the most recent attrs — released no less than a week ago! — which seems to support this, (2) unslot this class, or (3) handroll our own slotted cached property? Which I don't think would be hard, but could maybe be very hard. 

I don't really like either (1) or (2). But I'm not also familiar with how much of a difference this makes on memory pressure.

Reply:
I think this is actually less of a concern than I was worried about. We don't call it that often, and the code is really fast, except potentially floats and strings. The only case I would be concerned about is traversing the same tree in a tight loop and calling this function many times. Let's leave it as a future potential optimization?